### ISSUE TITLE
double done?

### ISSUE THREAD

**Issue by volodymyrss** [oda-hub/frontend-astrooda#33]
![image](https://user-images.githubusercontent.com/3909535/120482082-12626c80-c3b1-11eb-9b0a-ff79b3f3db82.png)

**Comment by burnout87** [comment]
Just to confirm, did it also send a double `done` email? It should not...

**Comment by volodymyrss** [comment]
nope

**Comment by burnout87** [comment]
I have been trying to reproduce the bug quite un-seccussefully...do you remember some details of the request you did?

From the screenshot it looks like a simple request

**Comment by volodymyrss** [comment]
It's hard to say. It might be possible to understand from the code if this can even happen in principle.

**Comment by burnout87** [comment]
Got something similar:

![image](https://user-images.githubusercontent.com/3757833/121718235-61f81500-cae2-11eb-907e-8d3113f966ce.png)

**Comment by volodymyrss** [comment]
> Got something similar:
> 
> And it looks like the problem that two messages are sent from the backend...?
> 
> ![image](https://user-images.githubusercontent.com/3757833/121718235-61f81500-cae2-11eb-907e-8d3113f966ce.png)
ok!

please clarify: where did you get it: which dispatcher, which request. 
Also, you mean actually dda backend?
Or do you mean dispatcher (which is occasionally called backend in because of it's backend-y relation to frontend)?

**Comment by volodymyrss** [comment]
It's good to clarify those points.
But I am extra confused why frontend shows two messages of the kind which should be only shown at the end of the interaction with backends. I.e. frontend should not continue asking dispatcher after first message like that.
Beware that dispatcher does not actively send anything to frontend  - it does not initiate connections. It is frontend which is asking.

I just suspect it might be something like:
* request 1 was initiated; progress window shown
* progress window for request 1 closed by user
* request 2 initiated; progress window shown
* when request 1 is done, `done` (or fail) is shown. progress window remains
* when request 2 is done, `done` (or fail) is shown. it's shown in the same progress window
![image](https://user-images.githubusercontent.com/3909535/121720722-b780f180-cae3-11eb-8063-2ddebd39146b.png)

**Comment by burnout87** [comment]
> please clarify: where did you get it: which dispatcher, which request.
The request was sent to the  dispatcher internal release. 
So, I was NOT  connected to the VPN (I simply forgot to do it), and that is the reason why my request never went through

> Also, you mean actually dda backend?
> Or do you mean dispatcher (which is occasionally called backend in because of it's backend-y relation to frontend)?

I mean the dispatcher, sorry I was not specific.

**Comment by burnout87** [comment]
> It's good to clarify those points.
> But I am extra confused why frontend shows two messages of the kind which should be only shown at the end of the interaction with backends. I.e. frontend should not continue asking dispatcher after first message like that.
> Beware that dispatcher does not actively send anything to frontend - it does not initiate connections. It is frontend which is asking.

Yes, that is indeed the reason of my suspicions, multiple requests sent from the frontend

> 
> I just suspect it might be something like:
> 
>     * request 1 was initiated; progress window shown
> 
>     * progress window for request 1 closed by user
> 
>     * request 2 initiated; progress window shown
> 
>     * when request 1 is done, `done` (or fail) is shown. progress window remains
> 
>     * when request 2 is done, `done` (or fail) is shown. it's shown in the same progress window

Exactly! I am suspecting the same! It can be reproduced by not closing the progress window

>       ![image](https://user-images.githubusercontent.com/3909535/121720722-b780f180-cae3-11eb-8063-2ddebd39146b.png)

**Comment by burnout87** [comment]
Just reproduced again, just with a different error, but the sequence of steps was exactly the one we suspect:

![image](https://user-images.githubusercontent.com/3757833/121721149-370ec080-cae4-11eb-852f-321004b23788.png)

**Comment by volodymyrss** [comment]
> > please clarify: where did you get it: which dispatcher, which request.
> > The request was sent to the  dispatcher internal release.
> > So, I was NOT  connected to the VPN (I simply forgot to do it), and that is the reason why my request never went through
> 
> > Also, you mean actually dda backend?
> > Or do you mean dispatcher (which is occasionally called backend in because of it's backend-y relation to frontend)?
> 
> I mean the dispatcher, sorry I was not specific.

no problem! I only strive for specificity when we agree it benefits our goals.

Btw, it's all cool, and it is great if it's fixed, but really we have critical issues:

https://github.com/oda-hub/frontend-astrooda/issues?q=is%3Aopen+is%3Aissue+label%3Abug+milestone%3Av21.06

We can continue fixing issues as time goes, will make frequent fixes!

**Comment by burnout87** [comment]
I was actually working on another fix, and this came out...so it was good that it could be spotted and noted here.

**Comment by burnout87** [comment]
I did some more tests, and didn't reproduce the behavior here discussed.

I think we can close this for the moment, and re-open it in case it happens again.

**Comment by volodymyrss** [comment]
It happens again, in similar situation:

![image](https://user-images.githubusercontent.com/3909535/124878263-3296d400-dfcc-11eb-822b-4d68d3ecf483.png)

It could be that the difference is that it the request was not asynchronous. This is not a critical issue, but to keep track.
